### PR TITLE
Add spell checking to CI

### PR DESCRIPTION
**Pull request by Noah-Silvera**

### CODE REVIEW

**Comment by ianschmitz** [comment]
Maybe we want to check out https://www.npmjs.com/package/cspell? Could integrate with CI if it works well.

**Comment by ianschmitz** [comment]
We should probably wait to get the Web PR in first.

File: src/pages/index.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ const products = [
         title: "Geocortex Web",
         imageUrl: "img/web-viewer-product-icon.svg",
         description:
-            "Combine the 3D and 2D capabilities of Esri’s latest JavaScript API to deliver cutting-edge mapping applications.",
+            "Combine the 3D and 2D capabilities of Esris latest JavaScript API to deliver cutting-edge mapping applications.",
```
Comment (eric-parsons) [inline src/pages/index.js:14]: This one is a mistake. Needs to be reverted and the override removed from cSpell settings.

Comment (Noah-Silvera) [reply src/pages/index.js]: reverted,

**Comment by Noah-Silvera** [comment]
> Maybe we want to check out https://www.npmjs.com/package/cspell? Could integrate with CI if it works well. Would complement our existing `lint` script nicely IMO.

@ianschmitz Good call, I've updated to do that. I missed that package! Thanks :)

File: docs/workflow/implement-custom-activity-server.mdx [hunk 2]
```diff
@@ -46,7 +46,7 @@ namespace App1.Workflow.CustomActivity
 ```
 
 :::tip
-Use a unique prefix on the `Action` property to avoid it conflicting with someone else's custom activities.  
+Use a unique prefix on the `Action` property to avoid it conflicting with someone elses custom activities.
```
Comment (ianschmitz) [inline docs/workflow/implement-custom-activity-server.mdx:49]: I don't think this change was correct.

File: package.json [hunk 3]
```diff
@@ -5,7 +5,9 @@
   "scripts": {
     "start": "docusaurus start",
     "build": "docusaurus build",
-    "lint": "remark -f -q -u validate-links \"docs/**/*.mdx\" && npm run prettier -- --check",
+    "lint": "npm run validate-links && npm run spellcheck && npm run prettier -- --check",
+    "spellcheck": "cspell \"{docs,src}/**\"",
```
Comment (ianschmitz) [inline package.json:9]: If you want to split these out can you instead use `lint:links`, `lint:spelling`?

Comment (Noah-Silvera) [reply package.json]: Totally, done.

File: docs/workflow/implement-custom-activity-server.mdx (hunk 2 above)
Comment (Noah-Silvera) [reply docs/workflow/implement-custom-activity-server.mdx]: It wasn't.... the spellchecker was mistaken, it told me it was.
reverted and added an exception to the spellchecker.

**Comment by ianschmitz** [comment]
Pushed a few commits. Take a peek and let me know what you think.

**Comment by Noah-Silvera** [comment]
> Pushed a few commits. Take a peek and let me know what you think.

Looks good to me, except I needed to add back in the word `rerender`, else I was still getting cspell errors. for instances of `useWatchAndRerender`

**Pull request merged**